File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var findSpecialInteger = function(arr) {
    if (arr.length === 1) return arr[0];
    let count = 1;
    for (let i = 1; i < arr.length; i++) {
        if (arr[i] === arr[i-1]) {
            count++;
        } else {
            count = 1;
        }
        if (count > arr.length/4) return arr[i];
    }
};

File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.cpp
class Solution {
public:
    int findSpecialInteger(vector<int>& arr) {
        map<int,int>mp;
        int f;
        for(int i=0;i<arr.size();i++){
            mp[arr[i]]++;
        }
        int mx=0;
        for(auto x: mp){
            mx=max(mx,x.second);
        }
        for(auto x: mp){
            if(mx==x.second){ f=x.first; break; }
        }
            return f;
    }
};